dotCMS/src/main/java/com/dotmarketing/portlets/rules/RuleComponentDefinition.java

```diff
@@ -120,5 +138,41 @@ public String toLogString(){
     }
 
     public abstract boolean evaluate(HttpServletRequest request, HttpServletResponse response, T instance);
-}
 
+    /**
+     * Utility type used to correctly read immutable object from JSON representation.
+     *
+     * @deprecated Do not use this type directly, it exists only for the <em>Jackson</em>-binding
+     * infrastructure
+     */
+    @Deprecated
+    @JsonDeserialize
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
+    @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.NONE)
+    static protected final class Json {
+
+        @javax.annotation.Nullable
+        public String id;
+        @javax.annotation.Nullable
+        public String i18nKey;
+        @javax.annotation.Nullable
+        public Map<String, ParameterDefinition> parameterDefinitions;
+
+        @JsonProperty("id")
+        public void setId(@Nullable String id) {
+            this.id = id;
+        }
+
+        @JsonProperty("i18nKey")
+        public void setI18nKey(@Nullable String i18nKey) {
+            this.i18nKey = i18nKey;
+        }
+
+        @JsonProperty("parameterDefinitions")
+        public void setParameterDefinitions(
+                @Nullable Map<String, ParameterDefinition> parameterDefinitions) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'parameterDefinitions' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4925354)